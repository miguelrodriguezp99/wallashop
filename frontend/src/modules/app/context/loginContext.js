import { createContext, useState } from "react";

// Este es el que tenemos que consumir
export const LogginContext = createContext();

// Este es el que nos provee de acceso al contexto
export function LogginProvider({ children }) {
	const [token, setToken] = useState(null);
	const [image, setImage] = useState(null);
	const [user, setUser] = useState({});

	return (
		<LogginContext.Provider
			value={{
				token,
				setToken,
				image,
				setImage,
				user,
				setUser,
			}}
		>
			{children}
		</LogginContext.Provider>
	);
}
