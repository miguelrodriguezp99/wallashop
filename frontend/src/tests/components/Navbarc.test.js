import React from 'react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import Navbar from '../../modules/app/components/Navbarc';
import { LogginContext } from "../../modules/app/context/loginContext"; // Asegúrate de usar la ruta correcta
import { MemoryRouter } from "react-router-dom";
import {render, fireEvent, screen} from '@testing-library/react';

const fakeLogginContext = {
    token: 'testToken',
    user: 'id'
};

// Reemplazar useNavigate por una versión simulada en el dom (tambien simulado)
// Permite rastrear su comportamiento
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: jest.fn(),
}));

// Se hace para que no se ejecuten los modulos reales del codigo, sino que se simulan
jest.mock('../../modules/app/components/modals/userModal', () => () => <div>UserModal Mock</div>);  //Ahora la función userModal devuelve <div>UserModal Mock</div>
jest.mock('../../modules/app/components/modals/notificationModal', () => () => <div>NotificationModal Mock</div>); // Lo mismo para notificationModal

describe('Navbar component', () => {
  it('matches the snapshot', () => {
    const tree = renderer.create(
        <MemoryRouter>
          <LogginContext.Provider value={fakeLogginContext}>
            <Navbar />
          </LogginContext.Provider>
        </MemoryRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('handles navigation on click', () => {
    const navigateMock = jest.fn();
    require('react-router-dom').useNavigate.mockReturnValue(navigateMock);

    render( 
        <MemoryRouter>
            <LogginContext.Provider value={fakeLogginContext}>
                <Navbar />
            </LogginContext.Provider>
        </MemoryRouter>);

    fireEvent.click(screen.getByText(/wallashop/i));
    expect(navigateMock).toHaveBeenCalledWith('/wallashop/');
    
  });

  it('handles navigation on login button click', () => {
    const navigateMock = jest.fn();
    require('react-router-dom').useNavigate.mockReturnValue(navigateMock);

    render(
      <MemoryRouter>
        <LogginContext.Provider value={{ fakeLogginContext }}>
          <Navbar />
        </LogginContext.Provider>
      </MemoryRouter>
    );

    fireEvent.click(screen.getByText(/Login/i));
    expect(navigateMock).toHaveBeenCalledWith('/wallashop/login');
  });

  it('handles navigation on logout button click', () => {
    const navigateMock = jest.fn();
    require('react-router-dom').useNavigate.mockReturnValue(navigateMock);

    render(
      <MemoryRouter>
        <LogginContext.Provider value={{ token: 'testToken', setToken: jest.fn(), setImage: jest.fn() }}>
          <Navbar />
        </LogginContext.Provider>
      </MemoryRouter>
    );

    fireEvent.click(screen.getByText('Log out'));
    expect(navigateMock).toHaveBeenCalledWith('/wallashop');
  });

  it('handles navigation on signup button click', () => {
    const navigateMock = jest.fn();
    require('react-router-dom').useNavigate.mockReturnValue(navigateMock);

    render(
      <MemoryRouter>
        <LogginContext.Provider value={{fakeLogginContext}}>
          <Navbar />
        </LogginContext.Provider>
      </MemoryRouter>
    );

    fireEvent.click(screen.getByText(/Sign Up/i));
    expect(navigateMock).toHaveBeenCalledWith('/wallashop/register');
  });

});