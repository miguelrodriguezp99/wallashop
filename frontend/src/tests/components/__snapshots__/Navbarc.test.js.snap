// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component matches the snapshot 1`] = `
<nav
  className="flex z-40 w-full h-auto items-center justify-center data-[menu-open=true]:border-none sticky top-0 inset-x-0 backdrop-blur-lg data-[menu-open=true]:backdrop-blur-xl backdrop-saturate-150 bg-background/70"
  style={
    Object {
      "--navbar-height": "4rem",
    }
  }
>
  <header
    className="z-40 flex px-6 gap-4 w-full flex-row relative flex-nowrap items-center justify-between h-[var(--navbar-height)] max-w-[1024px]"
  >
    <div
      className="flex basis-0 flex-row flex-grow flex-nowrap justify-start bg-transparent items-center no-underline text-medium whitespace-nowrap box-border"
    >
      <p
        className="font-bold text-2xl cursor-pointer"
        onClick={[Function]}
      >
        Wallashop
      </p>
    </div>
    <ul
      className="flex gap-4 h-full flex-row flex-nowrap items-center data-[justify=start]:justify-start data-[justify=start]:flex-grow data-[justify=start]:basis-0 data-[justify=center]:justify-center data-[justify=end]:justify-end data-[justify=end]:flex-grow data-[justify=end]:basis-0"
      data-justify="end"
    >
      <li
        className="text-medium whitespace-nowrap box-border list-none data-[active=true]:font-semibold"
      >
        <div
          className="flex gap-2 items-center"
        >
          <div>
            NotificationModal Mock
          </div>
          <button
            className="group tap-highlight-transparent transition-transform-colors-opacity data-[hover=true]:opacity-hover z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-unit-4 min-w-unit-20 h-unit-10 text-small gap-unit-2 rounded-medium [&>svg]:max-w-[theme(spacing.unit-8)] data-[pressed=true]:scale-[0.97] transition-transform-colors-opacity motion-reduce:transition-none bg-primary/20 data-[hover=true]:opacity-hover text-black"
            disabled={false}
            href="#"
            onBlur={[Function]}
            onClick={[Function]}
            onDragStart={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchCancel={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex={0}
            type="button"
          >
            Log out
          </button>
          <div>
            UserModal Mock
          </div>
        </div>
      </li>
    </ul>
  </header>
</nav>
`;
